go/state/state_db.go

```diff
@@ -204,6 +206,10 @@ type stateDB struct {
 	// Tracks the clearing state of individual accounts.
 	clearedAccounts map[common.Address]accountClearingState
 
+	// A list of contracts created during the current transaction. This is used
+	// to mark eligible accounts to be self-destructed according to EIP-6780.
+	createdContracts map[common.Address]bool
```

Comment:
Since the `bool` value type has no semantic interpretation consider using an empty struct instead to stress the fields interpretation as a set:
```suggestion
	createdContracts map[common.Address]struct{}
```